assets/javascripts/cookies.js

```diff
@@ -1,43 +1,110 @@
-    banner.setAttribute('tabindex', '-1')
-    banner.focus()
+function showNotificationBanner() {
+    const notificationBanner = document.getElementById('notification-banner');
+    notificationBanner.hidden = false;
+    notificationBanner.setAttribute('tabindex', '-1');
+    notificationBanner.focus();
 
-    banner.addEventListener('blur', function () {
-        banner.removeAttribute('tabindex')
-    })
+    notificationBanner.addEventListener('blur', function() {
+        notificationBanner.removeAttribute('tabindex');
+    });
 }
 
-acceptButton.addEventListener('click', function (event) {
-    showBanner(acceptedBanner)
-    event.preventDefault()
-})
+function init() {
+
+    const acceptButton = document.querySelector('.js-cookies-button-accept');
+    const rejectButton = document.querySelector('.js-cookies-button-reject');
 
-rejectButton.addEventListener('click', function (event) {
-    showBanner(rejectedBanner)
-    event.preventDefault()
-})
+    const acceptedBanner = document.querySelector('.js-cookies-accepted');
+    const rejectedBanner = document.querySelector('.js-cookies-rejected');
+    const questionBanner = document.querySelector('.js-question-banner');
+    const cookieBanner = document.querySelector('.js-cookies-banner');
 
-acceptedBanner.querySelector('.js-hide').addEventListener('click', function () {
-  cookieBanner.setAttribute('hidden', 'hidden')
-})
+    function showBanner(banner) {
+        questionBanner.setAttribute('hidden', 'hidden');
+        banner.removeAttribute('hidden');
 
-rejectedBanner.querySelector('.js-hide').addEventListener('click', function () {
-  cookieBanner.setAttribute('hidden', 'hidden')
-})
+        banner.setAttribute('tabindex', '-1');
+        banner.focus();
+
+        banner.addEventListener('blur', function () {
+            banner.removeAttribute('tabindex');
+        });
+    }
+
+    acceptButton.addEventListener('click', function (event) {
+        showBanner(acceptedBanner);
+        updateCookieAnalyticsSettings(true);
+        setCookie('cookie_preferences', 'true', 365);
+        event.preventDefault();
+    });
+
+    rejectButton.addEventListener('click', function (event) {
+        showBanner(rejectedBanner);
+        updateCookieAnalyticsSettings(false);
+        setCookie('cookie_preferences', 'true', 365);
+        event.preventDefault();
+    });
+
+    acceptedBanner.querySelector('.js-hide').addEventListener('click', function () {
+        cookieBanner.setAttribute('hidden', 'hidden');
+    });
+
+    rejectedBanner.querySelector('.js-hide').addEventListener('click', function () {
+        cookieBanner.setAttribute('hidden', 'hidden');
+    });
 }
 
-init()
\ No newline at end of file
+document.addEventListener('DOMContentLoaded', function() {
+    if (!getCookie('cookie_settings')) {
+        updateCookieAnalyticsSettings(false);
+    }
+    const form = document.getElementById('cookie-settings-form');
+
+    form.addEventListener('submit', function(event) {
```

Comment:
If you check the console on any page except `/cookies` there's an error `Uncaught TypeError: Cannot read properties of null (reading 'addEventListener')
    at HTMLDocument.<anonymous> (cookies.js:100:10)` because the cookie-settings-form form only exists on `/cookies`.

You could add a small function to check if you are on the cookies page to prevent it running.

```
function isOnCookiesPage() {
return window.location.pathname === '/cookies'
}
```